cpp/test/adapter-non-local.test.js

```diff
@@ -1,268 +1,671 @@
+    })
+
+    it('Should set the topic0 and chainId correctly', async () => {
+      const originChainId = no0x(ethers.zeroPadValue('0x02',32))
+      await adapter.contract.actions
+        .settopiczero([hexStringToBytes(originChainId), hexStringToBytes(evmTopicZero)])
+        .send(active(adapter.account))
+      
+      const emitterRow = adapter.contract.tables
+        .mappings(getAccountCodeRaw(adapter.account))
+        .getTableRow(BigInt('0x' + originChainId.slice(-16)))
+      
+      expect(emitterRow.chain_id).to.be.equal(originChainId)
+      expect(emitterRow.topic_zero).to.be.equal(evmTopicZero)
+    })
+
+    it('Should update the topic0 correctly', async () => {
+      const originChainId = evmOperationSamples.pegin.originChainId
+      await adapter.contract.actions
+        .settopiczero([hexStringToBytes(originChainId), hexStringToBytes(evmTopicZero)])
+        .send(active(adapter.account))
+      
+      const emitterRow = adapter.contract.tables
+        .mappings(getAccountCodeRaw(adapter.account))
+        .getTableRow(BigInt('0x' + originChainId.slice(-16)))
+      
+      expect(emitterRow.chain_id).to.be.equal(originChainId)
+      expect(emitterRow.topic_zero).to.be.equal(evmTopicZero)
+    })
+  })
+  
+  describe('adapter::swap', () => {
+    it('Should swap correctly', async () => {
+    
+    })
+  })
+
+  describe('adapter::settle', () => {
+    // -> TODO move under PAM tests
+    it('Should reject if adapter and token do not match', async () => {
+      const operation = evmOperationSamples.pegin
+      const metadata = evmMetadataSamples.pegin
+
+      const action = adapter.contract.actions
+        .settle([user, {...operation, token: wrongToken.bytes}, metadata])
+        .send(active(user))
+
+      await expectToThrow(action, errors.INVALID_TOKEN)
+    })
+
+    it('Should reject if context is not correct', async () => {
+      const operation = evmOperationSamples.pegin
+      const metadata = evmMetadataSamples.peginWithWrongContext
+
+      const action = adapter.contract.actions
+        .settle([user, operation, metadata])
+        .send(active(user))
+
+      await expectToThrow(action, errors.UNEXPECTED_CONTEXT)
+    })
+
+    it('Should reject if signature is invalid', async () => {
+      const operation = evmOperationSamples.pegin
+      const metadata = evmMetadataSamples.peginWithWrongSignature
+
+      const action = adapter.contract.actions
+        .settle([user, operation, metadata])
+        .send(active(user))
+
+      await expectToThrow(action, errors.INVALID_SIGNATURE)
+    })
+    // <- TODO move under PAM tests
+
+    it('Should settle the operation properly', async () => {
+      const operation = evmOperationSamples.pegin
+      const metadata = evmMetadataSamples.pegin
+      
+      const quantity = `${amounts.pegin} TST`
+      const xquantity = `${trimPrecision(amounts.pegin, fromEvmXerc20.precision)} XTST`
```

Comment:
It isn't clear where this `amounts` is coming from, I would rather create an utility function that computes the amount from the operation one